integrated_channels/xapi/models.py

```diff
@@ -39,6 +48,25 @@ class XAPILRSConfiguration(TimeStampedModel):
         null=False,
         help_text=_('Is this configuration active?'),
     )
+    auth_method = models.CharField(
+        max_length=16,
+        verbose_name="xAPI POST Authentication Method",
+        choices=XAPIAuthMethods.choices,
+        default=XAPIAuthMethods.HTTP_BASIC,
+        help_text=_('The Authentication Method to use when sending the xAPI data to the endpoint.')
+    )
+    auth_url = models.URLField(
+        blank=True,
+        null=True,
+        help_text=_("URL to use for authentication. Eg., Token URL for OAuth")
```

Comment:
Should this point to the LMS OAuth2 (e.g., `localhost:18000/oauth2/access_token`) or directly to the `edx-enterprise` service? It would be good to clarify this here.

Reply:
Ah. I think I have badly explained this PR. The OAuth URL would be from an external LRS service. For example, [edCast](https://docs.edcast.com/docs/xapi#authentication) uses `<instance-hostname>/api/lrs/v1/xapi/oauth2/token`.